zkevm-circuits/src/evm_circuit/execution/logs.rs

```diff
@@ -62,9 +60,12 @@ impl<F: Field> ExecutionGadget<F> for LogGadget<F> {
         // use call context's  callee address as contract address
         let contract_address = cb.call_context(None, CallContextFieldTag::CalleeAddress);
         let is_persistent = cb.call_context(None, CallContextFieldTag::IsPersistent);
+        cb.require_boolean("is_persistent is bool", is_persistent.expr());
+
         cb.condition(is_persistent.expr(), |cb| {
             cb.tx_log_lookup(
                 tx_id.expr(),
+                cb.curr.state.log_id.expr() + 1.expr(),
```

Comment:
the point is ineed somewhat different from raw spec that I want to emphasis,  I realize that only in next step log id will plus 1,  in current log step, the log id is still zero,  as we define zero log id meaning no log inserting( like in the begin tx resetting it to zero).   in order to make log id universal between current step and next step(copy step) for log lookup , seems to use next log id(curr.state.log_id + 1.) is better , glad to know if you have other opinion ! if no, I will update spec for this minor change .

Reply:
@ed255 spec update is here .https://github.com/privacy-scaling-explorations/zkevm-specs/pull/214

Reply:
Thanks for updating the spec!  The approach sounds good to me; and it also seems reasonable that the first `log_id` used is `1`, because we're doing this with all counter already (tx_id, rwc, etc.)